src/selectors/filtered-pipeline.js

```diff
@@ -0,0 +1,107 @@
+import { createSelector } from 'reselect';
+
+const getEdgeIDs = (state) => state.edge.ids;
+const getEdgeSources = (state) => state.edge.sources;
+const getEdgeTargets = (state) => state.edge.targets;
+const getFromNodes = (state) => state.filters.from;
+const getToNodes = (state) => state.filters.to;
+
+/**
+ * Selector to get all edges formatted as an array of objects with id, source, and target properties.
+ * @param {Object} state - The global state object.
+ * @returns {Array} An array of edge objects.
+ */
+const getEdges = createSelector(
+  [getEdgeIDs, getEdgeSources, getEdgeTargets],
+  (edgeIDs, edgeSources, edgeTargets) =>
+    edgeIDs.map((id) => ({
+      id,
+      source: edgeSources[id],
+      target: edgeTargets[id],
+    }))
+);
```

Comment:
Is there any reason this logic can't live within the `getEdgesByNode` selector? It doesn't look like you're using it anywhere else.